shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/auth/builder/loader/dialect/MySQLPrivilegeLoader.java

```diff
@@ -52,6 +52,7 @@
     @Override
     public Map<ShardingSphereUser, ShardingSpherePrivilege> load(final Collection<ShardingSphereUser> users, final DataSource dataSource) throws SQLException {
         Map<ShardingSphereUser, ShardingSpherePrivilege> result = new LinkedHashMap<>();
+        users.forEach(user -> result.put(user, new ShardingSpherePrivilege()));
```

Comment:
Does that means this user has no privileges in databases, right?

Reply:
```
private void fillGlobalPrivilege(final Map<ShardingSphereUser, ShardingSpherePrivilege> privileges, final ResultSet resultSet) throws SQLException {
        Optional<ShardingSphereUser> user = getShardingSphereUser(privileges, resultSet);
        if (user.isPresent()) {
            privileges.get(user.get()).getAdministrativePrivilege().getPrivileges().addAll(loadAdministrativePrivileges(resultSet));
            privileges.get(user.get()).getDatabasePrivilege().getGlobalPrivileges().addAll(loadDatabaseGlobalPrivileges(resultSet));
        }
    }
```
We analyzed this code will find in `getShardingSphereUser` method, if parameter `privileges` is empty it will returns empty, so `user.isPresent()` always false. `fillGlobalPrivilege` can't add any elements to `privileges`.The same problem in `fillSchemaPrivilege` and `fillTablePrivilege`.  but the parameter `privileges` is empty at first.

so I add the below code in the first to avoid this problem.
```
users.forEach(user -> result.put(user, new ShardingSpherePrivilege()));
```
@tristaZero Can you help me see if this is right？

Reply:
@tristaZero  It simply initializes an empty container to hold the data. I discovered this problem through test cases